### PR TITLE
[CI] Move 2, 3: Copy job SourceTests, LintModifiedExtensions from Travis to ADO

### PR DESCRIPTION
**Pull request by haroldrandom**
This is the 2nd and 3rd move for CI to migrate from Travis to ADO which [move job SourceTests to ADO](https://github.com/Azure/azure-cli-extensions/issues/1140) 

- Move 2: SourceTests for integration test (incremental testing)
- Move 3: LintOnModifiedExtensions

Respect #867

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -66,50 +66,48 @@ jobs:
...
-#     displayName: 'Use Python 2.7'
-#     inputs:
-#       versionSpec: 2.7
-#   - task: Bash@3
-#     displayName: "Tests Python 2.7"
-#     inputs:
-#       targetType: 'filePath'
-#       filePath: scripts/ci/test_source.sh
+- job: SourceTests
+  displayName: "Unittest Test, Build Test"
```
Comment (MyronFanQiu) [inline azure-pipelines.yml]: Does Unittest mean integration test?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -66,50 +66,48 @@ jobs:
...
+
+- job: LintModifiedExtensions
+  displayName: "CLI Linter on Modified Extensions"
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - task: UsePythonVersion@0
+      displayName: 'Use Python 3.6'
+      inputs:
+        versionSpec: 3.6
```
Comment (MyronFanQiu) [inline azure-pipelines.yml]: Is it intentional to remove 3.7 and add 3.6?

Comment (haroldrandom) [reply azure-pipelines.yml]: No.
Python 3.6 takes 50%+ market share, Python 3.7 takes 12%+.
Besides, we actually put faith on Python version is independent as it should be.

File: azure-pipelines.yml (hunk 1 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Great, changed.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -66,50 +66,48 @@ jobs:
...
+- job: SourceTests
+  displayName: "Integration Tests, Build Tests"
+  pool:
+    vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+      Python38:
+        python.version: '3.8'
```
Comment (jsntcy) [inline azure-pipelines.yml]: I'm curious why we support 3.6 and 3.8 here, while support 3.7 in other jobs

Comment (haroldrandom) [reply azure-pipelines.yml]: Python 3.6 takes 57%+ market share, Python 3.7 takes 10%- https://w3techs.com/technologies/details/pl-python/3

**Pull request merged**